pkg/controller/atlasdeployment/backup.go

```diff
@@ -100,11 +109,16 @@ func backupScheduleManagedByAtlas(ctx context.Context, atlasClient mongodbatlas.
 			operatorBS.Policies[0].ID = atlasBS.Policies[0].ID
 		}
 
-		isSame, err := backupSchedulesAreEqual(atlasBS, operatorBS)
+		result, err := backupSchedulesAreEqualOrDefault(atlasBS, operatorBS)
 		if err != nil {
+			return false, nil
+		}
+		switch result {
+		case bsEqual:
 			return true, nil
+		default:
+			return false, nil
 		}
```

Comment:
The logic is reversed here. If the contents are equal, then it's not managed by atlas and it should return false